internal/cephfs/volume.go

```diff
@@ -260,3 +274,31 @@ func purgeVolume(ctx context.Context, volID volumeID, cr *util.Credentials, volO
 
 	return nil
 }
+
+// checkSubvolumeHasFeature verifies if the referred subvolume has
+// the required feature.
+func checkSubvolumeHasFeature(ctx context.Context, feature string, volID volumeID, cr *util.Credentials, volOptions *volumeOptions) (bool, error) {
+	subVolFeatures := volOptions.Features
+	if len(subVolFeatures) == 0 {
+		info, err := getSubVolumeInfo(ctx, volOptions, cr, volID)
+		if err != nil {
+			// For older Ceph versions where 'features' was not returned in the info output,
+			// or `subvolume info` command is not supported,it is assumed that these do not
+			// support the intended features.
+			// The Purge operation should then continue without snapshot retention.
+			if errors.Is(err, ErrInvalidCommand) {
+				return false, nil
+			}
+			return false, err
+		}
+		subVolFeatures = info.Features
+	}
```

Comment:
This part of the code is not required. let the user fetch the subvolume features and call this function

Reply:
Added it as suggested by Shayam in https://github.com/ceph/ceph-csi/pull/1458#discussion_r485671411 . This will make the function fail safe even if voloptions.Features is not populated. Should I get rid of it?

Reply:
Yes please, as this would be a helper function we should check only passed feature is present in `volOptions.Features` array

Reply:
Okay, will do it.